features/reward_features.py

```diff
@@ -291,6 +291,46 @@ def _start_tracking_out(self):
             super()._start_tracking_out()
             self.reward.off()
 
+
+class ScoreRewards(traits.HasTraits):
+    '''
+    Add a "score" to the task that keeps track of the number of successful trials and
+    the time taken to complete them. The score is displayed after each reward.
+    '''
+    score_display_location = traits.Tuple((10, 0, 10), desc="Location to display the score (in cm)")
+    score_display_height = traits.Float(1, desc="Height of the score display (in cm)")
+    score_display_color = traits.Tuple((1, 1, 1, 1), desc="Color of the score display")
+    score_timed_state = traits.String("target", desc="State to display the score after")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.reportstats['Score'] = 0
+
+    def _start_reward(self):
+        if hasattr(super(), '_start_reward'):
+            super()._start_reward()
+        timed_state = None
+        idx = -1
+        while timed_state is None and -idx-1 < len(self.state_log):
+            if self.state_log[idx][0] == self.score_timed_state:
+                timed_state = 1 + (self.state_log[-1][1] - self.state_log[idx][1])
```

Comment:
If you need to add a number so that the time is never 0, maybe you can add .001 or something much smaller? I guess it's a bit arbitrary